pkg/apikey/apikey.go

```diff
@@ -17,9 +17,16 @@ import (
 
 // Metadata stores non-secret metadata about the API key.
 type Metadata struct {
-	Name          string   `json:"name"`
+	// ID is the unique identifier of the API key inside the Turnkey database.
+	ID string `json:"id"`
```

Comment:
Are you talking about the API user that the key is attached to? Or the authenticator ID?

And one precision: when you say "Turnkey Database" I imagine you mean "organization data"?

Reply:
This is the authenticator (apikey) ID. Though, now that you mention it, it would probably be a good idea to _also_ be able store the User ID.

Since this is an external-facing package, I didn't want to use the term "organization data", but yes, that is correct:  Turnkey Database <=> Organization Data.

Reply:
Added userID to metadata, too

Reply:
With this we're making the assumption that an API key is only tied to a single org (single "ID"). I think we need to make this an array to support the decently common case of API key reuse across multiple organizations?

Reply:
As the ID is a uuid, there should be no reason to use different IDs across organisations.  Am i missing something?

Reply:
Both the user ID and the apikey id should be fungible across organisations.
Indeed. I would prefer to index these outside of the affected organisations, too.

Reply:
Unfortunately that's not the case today: the same API key added to two different organizations will generate 2 different UUIDs. I agree with you that there is no good reason why: we could just as well use the same UUID. I think generating new random UUIDs was easier because enclaves, when processing activities, only have context on the organization they operate over. Not the whole set of Turnkey organizations.

Reply:
Alas, I'd have preferred just using the public key as the unique identifier, but that ship has sailed, and so, too, it seems, the fungibility of the IDs.

Okay, that is not going to be useful for my intended use case, then.

Thanks for the explanation.